plugin/core/completion.py

```diff
@@ -1,76 +1,61 @@
-        else:
-            debug("WARNING: Replacement prefix does not match trigger '{}'".format(trigger))
-
-    if len(replacement) > 0 and replacement[0] == '$':  # sublime needs leading '$' escaped.
-        replacement = '\\$' + replacement[1:]
-    # only return trigger with a hint if available
-    return "\t  ".join((trigger, hint)) if hint else trigger, replacement
-
-
-def text_edit_text(item: dict, word_col: int) -> Optional[str]:
-    text_edit = item.get('textEdit')
-    if text_edit:
-        edit_range, edit_text = text_edit.get("range"), text_edit.get("newText")
-        if edit_range and edit_text:
-            edit_range = Range.from_lsp(edit_range)
-
-            # debug('textEdit from col {}, {} applied at col {}'.format(
-            #     edit_range.start.col, edit_range.end.col, word_col))
-
-            if edit_range.start.col <= word_col:
-                # if edit starts at current word, we can use it.
-                # if edit starts before current word, use the whole thing and we'll fix it up later.
-                return edit_text
-
-    return None
+compleiton_kinds = {
+    1: (sublime.KIND_ID_MARKUP, "Ξ", "Text"),
+    2: (sublime.KIND_ID_FUNCTION, "λ", "Method"),
+    3: (sublime.KIND_ID_FUNCTION, "λ", "Function"),
+    4: (sublime.KIND_ID_FUNCTION, "c", "Constructor"),
+    5: (sublime.KIND_ID_VARIABLE, "f", "Field"),
+    6: (sublime.KIND_ID_VARIABLE, "v", "Variable"),
+    7: (sublime.KIND_ID_TYPE, "c", "Class"),
+    8: (sublime.KIND_ID_TYPE, "i", "Interface"),
+    9: (sublime.KIND_ID_NAMESPACE, "◪", "Module"),
+    10: (sublime.KIND_ID_VARIABLE, "ρ", "Property"),
+    11: (sublime.KIND_ID_VARIABLE, "u", "Unit"),
+    12: (sublime.KIND_ID_VARIABLE, "ν", "Value"),
+    13: (sublime.KIND_ID_TYPE, "ε", "Enum"),
+    14: (sublime.KIND_ID_KEYWORD, "κ", "Keyword"),
+    15: (sublime.KIND_ID_SNIPPET, "s", "Snippet"),
+    16: (sublime.KIND_ID_AMBIGUOUS, "c", "Color"),
+    17: (sublime.KIND_ID_AMBIGUOUS, "#", "File"),
+    18: (sublime.KIND_ID_AMBIGUOUS, "⇢", "Reference"),
+    19: (sublime.KIND_ID_AMBIGUOUS, "ƒ", "Folder"),
+    20: (sublime.KIND_ID_TYPE, "ε", "EnumMember"),
+    21: (sublime.KIND_ID_VARIABLE, "π", "Constant"),
+    22: (sublime.KIND_ID_TYPE, "s", "Struct"),
+    23: (sublime.KIND_ID_FUNCTION, "e", "Event"),
+    24: (sublime.KIND_ID_KEYWORD, "ο", "Operator"),
+    25: (sublime.KIND_ID_TYPE, "τ", "Type Parameter")
+}
+
+
+def format_completion(item: dict, word_col: int) -> sublime.CompletionItem:
+    trigger = item.get('label') or ""
+    annotation = item.get('detail') or ""
+    kind = sublime.KIND_AMBIGUOUS
+
+    item_kind = item.get("kind")
+    if item_kind:
+        kind = compleiton_kinds.get(item_kind, sublime.KIND_AMBIGUOUS)
+
+    is_deprecated = item.get("deprecated", False)
+    if is_deprecated:
+        list_kind = list(kind)
+        list_kind[1] = '⚠'
+        list_kind[2] = "⚠ {} - Deprecated".format(list_kind[2])
+        kind = tuple(list_kind)  # type: ignore
```

Comment:
mypy complained here that I tried to assign `Tuple[object, ...]` to `Tuple[int, str, str]`.

I tried to use `kind = cast(Tuple[int, str, str], tuple(list_kind))` here but sublime cant import cast from typings it, it results in a run-time error :)

so I put # type: ignore

is this ok?

Reply:
Sounds OK to me for now.